### DISCUSSION TITLE
Tauri v2 menu items not visible

### DISCUSSION THREAD

**Discussion by mainrs** [tauri-apps/tauri#10101] [Q&A]
I use the following code to register a custom menu as well as to listen to events.

```rust
 let builder = tauri::Builder::default()
        .manage(state)
        .menu(|handle| {
            Menu::with_items(
                handle,
                &[&Submenu::with_items(
                    handle,
                    "Edit",
                    true,
                    &[
                        &MenuItem::with_id(handle, "preferences-item", "Preferences", true, Some("CommandOrControl+,"))
                            .unwrap(),
                    ],
                )
                .unwrap()],
            )
        })
        .setup(|app| {
            app.on_menu_event(|_, event| {
                println!("{:?}", event);
            });

            Ok(())
        })
        .plugin(tauri_plugin_shell::init())
		// ...
```

However, the window only shows my web content. There is no menu bar visible.

I am running Linux (Manjaro) and GNOME. If that matters.

**Comment by Charlie284** [comment]
Your menu construction is valid. On Linux, `Builder::menu` is intended to attach that menu to each configured window and render it above the webview; there is no frontend step required.

Because this was posted while Tauri v2 was still in beta, I would first update and make sure the Rust `tauri`/`tauri-build` crates and the JS CLI/API packages are on the same current v2 release. `pnpm tauri info` (or the equivalent npm command) will show the resolved versions.

You can also bypass the app-wide startup attachment and test the GTK window attachment directly:

```rust
use tauri::{
  menu::{Menu, MenuItem, Submenu},
  Manager,
};

tauri::Builder::default()
  .manage(state)
  .setup(|app| {
    let preferences = MenuItem::with_id(
      app,
      "preferences-item",
      "Preferences",
      true,
      Some("CommandOrControl+,"),
    )?;
    let edit = Submenu::with_items(app, "Edit", true, &[&preferences])?;
    let menu = Menu::with_items(app, &[&edit])?;

    let window = app
      .get_webview_window("main")
      .expect("main window was not created");

    window.set_menu(menu)?;
    window.show_menu()?;

    app.on_menu_event(|_, event| println!("{event:?}"));
    Ok(())
  })
  .plugin(tauri_plugin_shell::init())
  // ...
```

If this works, the menu itself is fine and the problem was in the old beta/version combination or the startup attachment path. If it still does not appear, please include the output of `pnpm tauri info`, the `app.windows` section of `tauri.conf.json`, and whether the GNOME session is Wayland or X11. That is enough to distinguish a Tauri/GTK issue from a window configuration issue.

The current menu guide and API still document `Builder::menu` as the normal app-wide form: https://v2.tauri.app/learn/window-menu/ and https://docs.rs/tauri/latest/tauri/struct.Builder.html#method.menu
